source/particle/generator/probability_density_function.cc

```diff
@@ -0,0 +1,295 @@
+                next_cell_weight = accumulated_cell_weights.back();
+
+              for (unsigned int q = 0; q < n_quadrature_points; ++q)
+                next_cell_weight += std::fabs(function.value(position[q]) * fe_values.JxW(q));
+
+              // Start from the weight of the previous cell
+              accumulated_cell_weights.push_back(next_cell_weight);
+            }
+        return accumulated_cell_weights;
+      }
+
+      template <int dim>
+      std::multimap<types::LevelInd, Particle<dim> >
+      ProbabilityDensityFunction<dim>::generate_particles_in_subdomain (const std::map<double,types::LevelInd> &cells,
+                                                                        const double global_weight,
+                                                                        const double start_weight,
+                                                                        const types::particle_index num_particles,
+                                                                        const types::particle_index start_id)
+      {
+        std::multimap<types::LevelInd, Particle<dim> > particles;
+
+        // Pick cells and assign particles at random points inside them
+        for (types::particle_index cur_id = start_id; cur_id < start_id + num_particles; ++cur_id)
+          {
+            // Draw the random number that determines the cell of the tracer
+            const double random_weight =  global_weight * uniform_distribution_01(random_number_generator);
+
+            // If the selected cell is not owned by this process, continue with next tracer
+            const std::map<double,types::LevelInd>::const_iterator select_cell_iterator = cells.lower_bound(random_weight);
+            if ((random_weight < start_weight) || (select_cell_iterator == cells.end()))
+              continue;
+
+            const types::LevelInd select_cell = select_cell_iterator->second;
+
+            const typename parallel::distributed::Triangulation<dim>::active_cell_iterator
+            it (&(this->get_triangulation()), select_cell.first, select_cell.second);
+
+            Point<dim> max_bounds, min_bounds;
+            // Get the bounds of the cell defined by the vertices
+            for (unsigned int d=0; d<dim; ++d)
+              {
+                min_bounds[d] = INFINITY;
+                max_bounds[d] = -INFINITY;
+              }
+            for (unsigned int v = 0; v < GeometryInfo<dim>::vertices_per_cell; ++v)
+              {
+                const Point<dim> vertex_position = it->vertex(v);
+                for (unsigned int d=0; d<dim; ++d)
+                  {
+                    min_bounds[d] = fmin(vertex_position[d], min_bounds[d]);
+                    max_bounds[d] = fmax(vertex_position[d], max_bounds[d]);
+                  }
+              }
+
+            // Generate random points in these bounds until one is within the cell
+            unsigned int num_tries = 0;
```

Comment:
we pretty consistently use `n_` instead of `num_` throughout ASPECT and deal.II. Maybe search through all of the particle files and see if you can globally replace the latter by the former?